game/world/objects/npc.cpp

```diff
@@ -1345,29 +1348,48 @@ bool Npc::implGoTo(uint64_t dt, float destDist) {
   if(go2.flag==GT_No)
     return false;
 
-  if(isInAir()) {
+  if(isInAir() || interactive()!=nullptr) {
     mvAlgo.tick(dt);
     return true;
     }
 
-  auto dpos = go2.target()-position();
+  auto  dpos      = go2.target()-position();
+  float destDistY = 200.f;
 
   if(go2.flag==GT_Flee) {
     // nop
     }
   else if(mvAlgo.isClose(go2.target(),destDist)) {
+    if(std::abs(dpos.y)>destDistY && recalculateWayPath()) {
```

Comment:
Is it check for waypath pointing vertically to top, so ladder can be assumed?
Also `200` is somewhat arbitrary, in  theory it can be a small hill with `dpos = {1000, 200,0}`

Reply:
Ladder points are completely precomputed while loading. Every ladder is correctly associated with a waypoint or not except the one we talked above. 

There can be a problem if npc climbs ladder, pops waypoint and falls down because of a monster or bad terrain. Then walking is continued on the lower floor because y-value is not used in waypoint distance checking. Maybe there is a better way to achieve this?